lib/config-file/src/parameter_provider.rs

```diff
@@ -0,0 +1,105 @@
+use async_trait::async_trait;
+use config::{ConfigBuilder, Value};
+use serde::{Deserialize, Serialize};
+use std::error::Error;
+use tracing::warn;
+
+#[derive(Debug, thiserror::Error)]
+pub enum ParameterError {
+    #[error("Failed to fetch parameters: {0}")]
+    Fetch(String),
+    #[error(transparent)]
+    Other(#[from] Box<dyn Error + Send + Sync>),
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct Parameter {
+    pub name: String,
+    pub value: Option<String>,
+}
+
+#[async_trait]
+pub trait ParameterProvider: Send + Sync {
+    async fn get_parameters_by_path(&self, path: String) -> Result<Vec<Parameter>, ParameterError>;
+}
```

Comment:
Providing a trait here to get around cyclic dependencies as basically everything depends on this crate eventually. This will let us swap out clients later, too, in case we ever need to do that. Neat.